### ISSUE TITLE
React packager does not reload required HTML file for WebView

### ISSUE THREAD

**Issue by sodik82** [facebook/react-native#10738]
### Description

I have simple HTML file asset and are using `WebView` to display it. Works fine for first time but when I reload simulator with `cmd-r` the HTML file is not reloaded when I change it.

```
        <WebView
          style={{width: 300}}
          source={require('./html/index.html')}
          javaScriptEnabled={true}
          />
```

logs from packager:
```
[11/4/2016, 10:56:31 AM] <END>   Initializing Packager (559ms)
[11/4/2016, 10:56:35 AM] <START> Requesting bundle: {"url":"/index.ios.bundle?platform=ios&dev=true&minify=false"}
[11/4/2016, 10:56:35 AM] <START> Transforming modules
transformed 644/644 (100%)
[11/4/2016, 10:56:35 AM] <END>   Transforming modules (353ms)
[11/4/2016, 10:56:35 AM] <END>   Requesting bundle: {"url":"/index.ios.bundle?platform=ios&dev=true&minify=false"} (466ms)
[11/4/2016, 10:56:36 AM] <START> Processing asset request: {"asset":"html/index.html"}
[11/4/2016, 10:56:36 AM] <END>   Processing asset request: {"asset":"html/index.html"} (4ms)
[11/4/2016, 10:56:46 AM] <START> Requesting bundle: {"url":"/index.ios.bundle?platform=ios&dev=true&minify=false"}
[11/4/2016, 10:56:46 AM] <END>   Requesting bundle: {"url":"/index.ios.bundle?platform=ios&dev=true&minify=false"} (28ms)
[11/4/2016, 10:57:09 AM] <START> Requesting bundle: {"url":"/index.ios.bundle?platform=ios&dev=true&minify=false"}
[11/4/2016, 10:57:09 AM] <START> Transforming modules
transformed 644/644 (100%)
[11/4/2016, 10:57:09 AM] <END>   Transforming modules (366ms)
[11/4/2016, 10:57:10 AM] <END>   Requesting bundle: {"url":"/index.ios.bundle?platform=ios&dev=true&minify=false"} (537ms)
```
as we can see html is served only once. HTML is reloaded (once) after manually restarting packager.

### Additional Information

* React Native version: "react-native": "0.36.1"
* Platform: iOS (android not tested)
* Operating System: MacOS

**Comment by brentvatne** [comment]
I'm not sure if this is related to the packager, it seems like the WebView doesn't request it a second time because it is cached.

**Comment by sodik82** [comment]
also possible but why restart of packager helps? I didn't want to mention it - but in my "required" HTML I included CSS with relative path and it seemed to be cached (even across) packager restarts. But I didn't want to "complicate" this issue even more. 
Unfortunately I haven't found any "caching" property of WebView in the documentation.

**Comment by brentvatne** [comment]
@sodik82 - you could try clearing your simulator cache to see if it's the UIWebView cache or something else: http://stackoverflow.com/a/36320592/659988

**Comment by grabcode** [comment]
I reckon the issue might concern any assets as a matter of fact.

It seems to be due to the latest changes made on the [AssetServer](https://github.com/facebook/react-native/blob/cacc31d7596acb5508aec617618eef432dd80595/packager/react-packager/src/AssetServer/index.js). It used to work before, and then I was using RN 0.34.0.

I'm currently working around the issue editing the file `node_modules/react-native/packager/react-packager/src/AssetServer/index.js`:
```javascript
_onFileChange(type, assetPath, file) {
    const asset = this._files.get(assetPath);
    this._hashes.delete(assetPath);
  }
```

Once I've pin pointed the solution, I'll share it around, unless anyone comes first ;).

**Comment by sodik82** [comment]
@brentvatne I tried following: started simulator, change the HTML file, reload sim -> no change, delete `~/Library/Caches`, reload sim -> still no change.

**Comment by grabcode** [comment]
@sodik82 did you get a chance to try out the solution I proposed? It's fairly quick to implement and test, and should demonstrate the issue has nothing to do with the UIWebView.

**Comment by sodik82** [comment]
@grabcode - yes, your PR (after removing logger - most likely it was not yet in 0.36) seems to work now.

**Comment by grabcode** [comment]
Thanks for testing @sodik82. You are correct, in RN 0.36.x, Packager used to employ a module called `Activity`, but is currently being refactored to use `Logger`. A bit of overhead whilst writing the PR but also an opportunity to get the AssetServer casting some light over the hidden intricity it does..

**Comment by dannycochran** [comment]
Possibly related, but I'm noticing that any js files loaded in the webview.html, e.g.

`
<script src="js/localJsFile.js"></script>
`

do not get updated on change. that's with the fix proscribed above for onFileChange.

**Comment by grabcode** [comment]
@dannycochran I've trouble check the issue you have, and it is not strictly due to the same reason. You're problem comes down `node_modules/react-native/packager/react-packager/src/Server/index.js`:
```javascript
res.setHeader('Cache-Control', 'max-age=31536000');
```

You may fix it yourself doing:
```javascript
if (process.env.NODE_ENV === 'production') {
  res.setHeader('Cache-Control', 'max-age=31536000');
}
```

I'm going to propose another PR on 0.37.x

Cheers,
Alex

**Comment by Zyphrax** [comment]
@grabcode I am having the same problem with React-Native 0.37 and I can confirm that replacing the `_onFileChange` with your code indeed solves the problem.

This is not an issue with the native app or the simulator, but with the React-Native asset system.

Sidenote: it would be great if I could simply tell the packager to include a folder with static assets (both in webserver and in bundle mode). I don't want to hard code these assets in the native app and to use them in the WebView component.

**Comment by dannycochran** [comment]
the onFileChange fix no longer works with  >= 0.39

would be great to close https://github.com/facebook/react-native/pull/10774

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with Canny! Check out its new home: https://react-native.canny.io/feature-requests/p/react-packager-does-not-reload-required-html-file-for-webview
